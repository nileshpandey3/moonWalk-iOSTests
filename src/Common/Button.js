import React from "react";
import styled from "styled-components";
import LinearGradient from "react-native-linear-gradient";
import Icon from "react-native-vector-icons/FontAwesome5";

const TouchableWrapper = styled.TouchableOpacity`
  shadow-opacity: 0.75;
  shadow-radius: 5px;
  shadow-color: ${({ shadow }) => shadow};
  shadow-offset: 0px 0px;
  margin: 0 5px;
  border-radius: 10px;
  background: ${({ theme }) => theme.cardBackground};
  ${({ disabled }) => disabled && "opacity: 0.3;"};
`;

const Wrapper = styled(LinearGradient)`
  border-radius: 10px;
  padding: 15px 30px;
  align-items: center;
  justify-content: center;
  flex-direction: row;
`;

const Title = styled.Text`
  color: #efefef;
  font-size: ${props => props.fontSize};
  font-weight: bold;
  margin-left: 10px;
  text-align: center;
`;

const TYPES = {
  primary: {
    gradient: ["#6b2afd", "#5e15ea"],
    shadow: "#5e15ea"
  },
  secondary: {
    gradient: ["#3a3a63", "#2c2f53"],
    shadow: "#2c2f53"
  },
  blue: {
    gradient: ["#6e90fe", "#5cd7cd"],
    shadow: "#6e90fe"
  },
  fire: {
    gradient: ["#ffec2a", "#fa53d6"],
    shadow: "#fa53d6"
  },
  red: {
    gradient: ["#FEB692", "#EA5455"],
    shadow: "#EA5455"
  }
};

export default ({
  title,
  onPress,
  type = "primary",
  disabled = false,
  icon,
  style,
  fontSize = 20
}) => (
  <TouchableWrapper
    style={style}
    onPress={onPress}
    shadow={TYPES[type].shadow}
    disabled={disabled}
  >
    <Wrapper
      colors={TYPES[type].gradient}
      start={{ x: 0.0, y: 0.0 }}
      end={{ x: 1.0, y: 0.0 }}
    >
      {icon && <Icon name={icon} size={25} color="#efefef" />}
      {title && <Title fontSize={fontSize}>{title}</Title>}
    </Wrapper>
  </TouchableWrapper>
);
