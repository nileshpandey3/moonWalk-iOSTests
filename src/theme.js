export default {
  cardBackground: "#42446f",
  inactive: "#8183a5"
};
