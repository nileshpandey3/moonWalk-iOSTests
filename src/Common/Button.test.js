import React from "react";
import Button from "./Button";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import theme from "../theme";

const renderWithTheme = component =>
  renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
describe("Button", () => {
  it("renders correctly", () => {
    const tree = renderWithTheme(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with title correctly", () => {
    const tree = renderWithTheme(<Button title="Press me" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with a different type style", () => {
    const tree = renderWithTheme(
      <Button type="secondary" title="Secondary" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
