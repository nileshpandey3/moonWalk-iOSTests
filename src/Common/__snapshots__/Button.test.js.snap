// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#42446f",
      "borderRadius": 10,
      "marginBottom": 0,
      "marginLeft": 5,
      "marginRight": 5,
      "marginTop": 0,
      "opacity": 1,
      "shadowColor": "#5e15ea",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.75,
      "shadowRadius": 5,
    }
  }
>
  <BVLinearGradient
    colors={
      Array [
        4285213437,
        4284356074,
      ]
    }
    endPoint={
      Object {
        "x": 1,
        "y": 0,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 10,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingBottom": 15,
          "paddingLeft": 30,
          "paddingRight": 30,
          "paddingTop": 15,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Button renders with a different type style 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#42446f",
      "borderRadius": 10,
      "marginBottom": 0,
      "marginLeft": 5,
      "marginRight": 5,
      "marginTop": 0,
      "opacity": 1,
      "shadowColor": "#2c2f53",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.75,
      "shadowRadius": 5,
    }
  }
>
  <BVLinearGradient
    colors={
      Array [
        4282006115,
        4281085779,
      ]
    }
    endPoint={
      Object {
        "x": 1,
        "y": 0,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 10,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingBottom": 15,
          "paddingLeft": 30,
          "paddingRight": 30,
          "paddingTop": 15,
        },
        undefined,
      ]
    }
  >
    <Text
      fontSize={20}
      style={
        Array [
          Object {
            "color": "#efefef",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginLeft": 10,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Secondary
    </Text>
  </BVLinearGradient>
</View>
`;

exports[`Button renders with title correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#42446f",
      "borderRadius": 10,
      "marginBottom": 0,
      "marginLeft": 5,
      "marginRight": 5,
      "marginTop": 0,
      "opacity": 1,
      "shadowColor": "#5e15ea",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.75,
      "shadowRadius": 5,
    }
  }
>
  <BVLinearGradient
    colors={
      Array [
        4285213437,
        4284356074,
      ]
    }
    endPoint={
      Object {
        "x": 1,
        "y": 0,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 10,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingBottom": 15,
          "paddingLeft": 30,
          "paddingRight": 30,
          "paddingTop": 15,
        },
        undefined,
      ]
    }
  >
    <Text
      fontSize={20}
      style={
        Array [
          Object {
            "color": "#efefef",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginLeft": 10,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Press me
    </Text>
  </BVLinearGradient>
</View>
`;
